File: server/src/main/java/edu/s_kuhn/prost/controller/TestController.java
package edu.s_kuhn.prost.controller;

import edu.s_kuhn.prost.persistance.entity.Test;
import edu.s_kuhn.prost.persistance.repository.TestRepository;
import java.util.UUID;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class TestController {
  private final TestRepository testRepository;

  public TestController(TestRepository testRepository) {
    this.testRepository = testRepository;
  }

  @GetMapping("/test/{id}")
  public String test(@PathVariable UUID id) {
    Test test =
        testRepository
            .findById(id)
            .orElseThrow(() -> new IllegalArgumentException("Test not found"));
    return test.getName();
  }
}

File: server/src/main/kotlin/edu/s_kuhn/prost/controller/TestController.kt
package edu.s_kuhn.prost.controller

import edu.s_kuhn.prost.persistence.repository.TestRepository
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController
import java.util.UUID

@RestController
class TestController(private val repository: TestRepository) {
    @GetMapping("/test/{id}")
    fun test(@PathVariable id: UUID): String {
        val test = repository.findById(id)
            .orElseThrow { IllegalArgumentException("Test not found") }
        return test.name
    }
}